File: classic puzzle-medium/primitive-pythagorean-triples.py
#https://www.codingame.com/training/medium/primitive-pythagorean-triples
#PRIMITIVE PYTHAGOREAN TRIPLES
import math
cnt=0
c=int(input())
#https://mathworld.wolfram.com/PythagoreanTriple.html
for n in range(1,math.ceil(c**0.5)):
    for m in range(n+1,math.ceil(c**0.5)):
        if math.gcd(m,n)==1 and (m+n)%2==1 and m*m+n*n<=c:
            cnt+=1
print(cnt)

File: classic puzzle-medium/primitive-pythagorean-triples.cpp
//https://www.codingame.com/training/medium/primitive-pythagorean-triples
//PRIMITIVE PYTHAGOREAN TRIPLES
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int C;
    cin>>C;
    int c=ceil(sqrt(C));
    int cnt=0;
    //https://mathworld.wolfram.com/PythagoreanTriple.html
    for(int m=1;m<c;m++)
        for(int n=m+1;n<c;n++)
            if(gcd(m,n)==1&&(m+n)%2==1&&(m*m+n*n)<=C) cnt++;
    cout<<cnt<<endl;
}
